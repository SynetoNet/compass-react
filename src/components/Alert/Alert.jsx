import React from "react";
import PropTypes from "prop-types";
import BSAlert from "react-bootstrap/Alert";
import "./Alert.scss";

class Alert extends React.Component {
  static Link = BSAlert.Link;
  static Heading = BSAlert.Heading;
  render() {
    return <BSAlert {...this.props} />;
  }
}

Alert.propTypes = {
  variant: PropTypes.oneOf([
    "primary",
    "secondary",
    "success",
    "danger",
    "warning",
    "info",
    "dark",
    "light"
  ]),
  dismissible: PropTypes.bool
};

Alert.defaultProps = {
  variant: "primary",
  dismissible: false
};

export default Alert;
