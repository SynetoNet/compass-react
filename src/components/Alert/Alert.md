#### **Default**

```jsx
[
  "primary",
  "secondary",
  "success",
  "danger",
  "warning",
  "info",
  "light",
  "dark"
].map((variant, idx) => (
  <Alert key={idx} variant={variant}>
    This is a {variant} alertâ€”check it out!
  </Alert>
));
```

#### **Links**

```jsx
[
  "primary",
  "secondary",
  "success",
  "danger",
  "warning",
  "info",
  "light",
  "dark"
].map((variant, idx) => (
  <Alert key={idx} variant={variant}>
    This is a {variant} alert with{" "}
    <Alert.Link href="#">an example link</Alert.Link>. Give it a click if you
    like.
  </Alert>
));
```

#### **Additional content**

```jsx
import { Icon } from "../../index.js";
<Alert variant="primary">
  <Alert.Heading>
    Hey, nice to see you! <Icon name="fas fa-user" />
  </Alert.Heading>

  <p>
    Aww yeah, you successfully read this important alert message. This example
    text is going to run a bit longer so that you can see how spacing within an
    alert works with this kind of content.
  </p>
  <hr />
  <p className="mb-0">
    Whenever you need to, click this <Alert.Link href="#">link</Alert.Link>.
  </p>
</Alert>;
```

#### **Dismissing**

```jsx
import { Button } from "../../index.js";
class AlertDismissibleExample extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      show: true
    };
  }

  render() {
    const handleDismiss = () => this.setState({ show: false });
    const handleShow = () => this.setState({ show: true });
    if (this.state.show) {
      return (
        <Alert variant="danger" onClose={handleDismiss} dismissible>
          <Alert.Heading>Oh snap! You got an error!</Alert.Heading>
          <p>
            Change this and that and try again. Duis mollis, est non commodo
            luctus, nisi erat porttitor ligula, eget lacinia odio sem nec elit.
            Cras mattis consectetur purus sit amet fermentum.
          </p>
        </Alert>
      );
    }
    return <Button onClick={handleShow}>Show Alert</Button>;
  }
}

<AlertDismissibleExample />;
```
