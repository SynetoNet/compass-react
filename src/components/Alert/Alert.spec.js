import React from "react";
import Alert from "./Alert";
import renderer from "react-test-renderer";

import { configure, mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Default Alert", () => {
  const defaultExample = (
    <Alert variant={"primary"}>This is a primary alertâ€”check it out!</Alert>
  );

  test("popover snapshop", () => {
    const component = shallow(defaultExample);
    expect(component).toMatchSnapshot();
  });
});
