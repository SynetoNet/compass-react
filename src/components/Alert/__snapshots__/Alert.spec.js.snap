// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Alert popover snapshop 1`] = `
<Alert
  closeLabel="Close alert"
  dismissible={false}
  show={true}
  transition={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "appear": false,
        "in": false,
        "mountOnEnter": false,
        "timeout": 300,
        "unmountOnExit": false,
      },
      "displayName": "Fade",
      "render": [Function],
    }
  }
  variant="primary"
>
  This is a primary alertâ€”check it out!
</Alert>
`;
